### PR TITLE
fix: Reintroduce babel transcompiler

### PR DESCRIPTION
**Pull request by dplewis**
### New Pull Request Checklist

- [ ] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server-push-adapter/security/policy).
- [ ] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server-push-adapter/issues?q=is%3Aissue).

### Issue Description

The server isn't compatible with ES6 modules. This PR transcompiles the project to CommonJS. In the future we could look into rollup as we might have to do this with the JS SDK.

https://github.com/parse-community/parse-server/actions/runs/9831140057/job/27138118586?pr=9182

### Approach

* Tested in Production
* Fixes Runtime Generator not found error
* Runs tests against `lib` instead of `src`
* Use nyc for coverage
* Use latest version of Babel

### CODE REVIEW

File: babel.config.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "presets": [
+    [
+      "@babel/preset-env",
+      {
+        "targets": {
+          "node": "14"
```
Comment (mtrezza) [inline babel.config.json:7]:
```suggestion
          "node": "18"
```

Should we bump this to the min. supported Node version?

**Comment by mtrezza** [comment]
See https://github.com/parse-community/parse-server/pull/9182#issuecomment-2213480255

**Comment by mtrezza** [comment]
So this can be closed thanks to https://github.com/parse-community/parse-server/pull/9182#issuecomment-2213847850, right?

**Comment by dplewis** [comment]
Yeah

**Pull request closed without merging**